### PR TITLE
Add HandwritingOCR SDK for windows platform 

### PR DESCRIPTION
**Pull request by Sinkee**
Add HandwritingOCR SDK for windows platform

### CODE REVIEW

File: ClientLibrary/Contract/HandwritingOCRLine.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+namespace Microsoft.ProjectOxford.Vision.Contract
+{
+    /// <summary>
+    /// The class of HandwritingOCR line.
+    /// </summary>
+    public class HandwritingOCRLine
```
Comment (qiwjin) [inline ClientLibrary/Contract/HandwritingOCRLine.cs]: double check the class name with xj

File: ClientLibrary/Contract/HandwritingOCRLine.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    {
+        /// <summary>
+        /// Gets or sets the bounding box.
+        /// </summary>
+        /// <value>
+        /// The bounding box.
+        /// </value>
+        public int[] BoundingBox { get; set; }
```
Comment (qiwjin) [inline ClientLibrary/Contract/HandwritingOCRLine.cs]: use int[] to represent box?

File: ClientLibrary/Contract/HandwritingOCROperation.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+namespace Microsoft.ProjectOxford.Vision.Contract
+{
+    /// <summary>
+    /// The class of HandwritingOCROperation
+    /// </summary>
+    public class HandwritingOCROperation
```
Comment (qiwjin) [inline ClientLibrary/Contract/HandwritingOCROperation.cs]: name?

File: ClientLibrary/Contract/HandwritingOCRWord.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    /// </summary>
+    public class HandwritingOCRWord
+    {
+        /// <summary>
+        /// Gets or sets the bounding box.
+        /// </summary>
+        /// <value>
+        /// The bounding box.
+        /// </value>
+        public int[] BoundingBox { get; set; }
```
Comment (qiwjin) [inline ClientLibrary/Contract/HandwritingOCRWord.cs]: same as above

File: Sample-WPF/HandwritingOCRPage.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        /// </summary>
+        /// <param name="imageUrl">The url to perform recognition on</param>
+        /// <param name="language">The language code to recognize for</param>
+        /// <returns></returns>
+        private async Task<HandwritingOCROperationResult> RecognizeUrl(string imageUrl)
+        {
+            return await Recognize(async (VisionServiceClient VisionServiceClient) => await VisionServiceClient.CreateHandwritingOCROperationAsync(imageUrl));
+        }
+
+        private async Task<HandwritingOCROperationResult> Recognize(Func<VisionServiceClient, Task<HandwritingOCROperation>> Func)
```
Comment (qiwjin) [inline Sample-WPF/HandwritingOCRPage.xaml.cs]: change the name to RecognizeAsync?

File: Sample-WPF/HandwritingOCRPage.xaml.cs [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+                result = await VisionServiceClient.GetHandwritingOCROperationResultAsync(operation);
+
+                int i = 0;
+                while ((result.Status == HandwritingOCROperationStatus.Running || result.Status == HandwritingOCROperationStatus.NotStarted) && i++ < MaxRetryTimes)
+                {
+                    Log(string.Format("Server status: {0}, wait {1} seconds...", result.Status, QueryWaitTimeInSecond));
+                    await Task.Delay(QueryWaitTimeInSecond);
+
+                    Log("Calling VisionServiceClient.GetHandwritingOCROperationResultAsync()...");
+                    result = await VisionServiceClient.GetHandwritingOCROperationResultAsync(operation);
```
Comment (qiwjin) [inline Sample-WPF/HandwritingOCRPage.xaml.cs]: add some code to demo how to use the result

**Review by qiwjin — CHANGES_REQUESTED** [review]

File: ClientLibrary/VisionServiceClient.cs [hunk 7]
```diff
@@ -664,7 +720,7 @@ private T ProcessAsyncResponse<T>(HttpWebResponse webResponse)
         private void SetCommonHeaders(WebRequest request)
         {
             request.ContentType = "application/json";
-            request.Headers[HttpRequestHeader.Authorization] = "Basic ZTkwNTE2ZmQ4NThlNDVjMmFhNDMzMjRlZjBlOThlN2E=";
+            //request.Headers[HttpRequestHeader.Authorization] = "Basic ZTkwNTE2ZmQ4NThlNDVjMmFhNDMzMjRlZjBlOThlN2E=";
```
Comment (hippogr) [inline ClientLibrary/VisionServiceClient.cs]: why remove this?

File: ClientLibrary/VisionServiceClient.cs [hunk 8]
```diff
@@ -715,6 +771,14 @@ private void HandleException(Exception exception)
...
 
                             ClientError errorCollection = JsonConvert.DeserializeObject<ClientError>(errorObjectString);
+
+                            if (errorCollection.Code == null && errorCollection.Message == null)
+                            {
+                                var errorType = new { Error = new ClientError() };
+                                var errorObj = JsonConvert.DeserializeAnonymousType(errorObjectString, errorType);
+                                errorCollection = errorObj.Error;
+                            }
+
```
Comment (hippogr) [inline ClientLibrary/VisionServiceClient.cs:782]: please add some comments to explain why we need do this

File: Sample-WPF/HandwritingOCRPage.xaml.cs [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+            HandwritingOCROperationResult result;
+            if (upload)
+            {
+                result = await UploadAndRecognizeImage(imageUri.LocalPath);
+            }
+            else
+            {
+                result = await RecognizeUrl(imageUri.AbsoluteUri);
+            }
+            _status.Text = "HandwritingOCR Done";
```
Comment (hippogr) [inline Sample-WPF/HandwritingOCRPage.xaml.cs]: handwriting recognition finished.

**Review by hippogr — CHANGES_REQUESTED** [review]

File: Sample-WPF/HandwritingOCRPage.xaml.cs (hunk 6 above)
Comment (Sinkee) [reply Sample-WPF/HandwritingOCRPage.xaml.cs]: We have a log result method, which is used to demo how to use the result.

File: ClientLibrary/Contract/HandwritingOCRLine.cs (hunk 1 above)
Comment (mike-int) [reply ClientLibrary/Contract/HandwritingOCRLine.cs]: Sync with xj and udpate.

File: ClientLibrary/Contract/HandwritingOCRLine.cs (hunk 2 above)
Comment (mike-int) [reply ClientLibrary/Contract/HandwritingOCRLine.cs]: keep the array and add a polygon to represent the line boundary.

File: ClientLibrary/Contract/HandwritingOCROperation.cs (hunk 3 above)
Comment (mike-int) [reply ClientLibrary/Contract/HandwritingOCROperation.cs]: update

File: ClientLibrary/Contract/HandwritingOCRWord.cs (hunk 4 above)
Comment (mike-int) [reply ClientLibrary/Contract/HandwritingOCRWord.cs]: keep the array and add a polygon to represent the word boundary.

File: Sample-WPF/HandwritingOCRPage.xaml.cs (hunk 5 above)
Comment (mike-int) [reply Sample-WPF/HandwritingOCRPage.xaml.cs]: update

File: ClientLibrary/VisionServiceClient.cs (hunk 7 above)
Comment (mike-int) [reply ClientLibrary/VisionServiceClient.cs]: recover

File: ClientLibrary/VisionServiceClient.cs (hunk 8 above)
Comment (mike-int) [reply ClientLibrary/VisionServiceClient.cs:782]: added.

File: Sample-WPF/HandwritingOCRPage.xaml.cs (hunk 9 above)
Comment (mike-int) [reply Sample-WPF/HandwritingOCRPage.xaml.cs]: update

**Review by qiwjin — APPROVED** [review]

**Pull request merged**